### ISSUE TITLE
Incidents reassigned to HUBOT_PAGERDUTY_FROM_EMAIL

### ISSUE THREAD

**Issue by xofer** [hubot-scripts/hubot-pager-me#125]
I may be confused about how this should work.

When I am on call, new incidents are assigned to me, naturally. But when I `misty pager ack`, they are reassigned to the user with `HUBOT_PAGERDUTY_FROM_EMAIL`. See below (our hubot is called misty):

![screenshot_20190102_141752](https://user-images.githubusercontent.com/1231804/50608061-54fa8480-0e99-11e9-94cd-c17739613edf.png)

According to the v1 to v2 migration docs, [requester should be specified in the From header](https://v2.developer.pagerduty.com/docs/migrating-to-api-v2#requester). It seems to me that this should be set to the user who's issuing the command (i.e. whatever I set with `misty pager me as ...`).

However, it's always set to the same value (`HUBOT_PAGERDUTY_FROM_EMAIL`). This poor guy gets reassigned every ticket.

https://github.com/hubot-scripts/hubot-pager-me/blob/414608b3e198b5dacaf5046ae1a9d25692646b57/src/pagerduty.coffee#L15-L19

Am I missing something here?

**Comment by xofer** [comment]
Based on experience and what's in the docs (linked above and quoted below), PD API v2 will reassign incidents to the user specified in the `From` header.

> In API v1, several endpoints provided a requester_id parameter, used to indicate what user the API client was acting on behalf of. This enabled an API client to acknowledge an incident on behalf of a user, invite a user on another user's behalf, etc.
>
> In API v2, this functionality is available using the HTTP From: header. Set this header with the email address of the user to act on behalf of.

**Comment by biggless** [comment]
There is possible problem when Slack email doesn't match PagerDuty email. Is it ok in such cases to fallback to static email from config?

**Comment by steve-stonehouse** [comment]
IMO it would be better for a message to be returned to the user saying that the email doesn't match. The API returns a `400` with a body of `{"error":{"message":"Requester User Not Found","code":1008}}` so should be possible.

**Comment by stephenyeargin** [comment]
@xofer @steve-stonehouse @biggless So is there a chance that #130 might address this?

**Comment by nobert** [comment]
I can confirm that #130 fixed this problem for us. (I forked this repo and merged a duplicate of #130 to my fork, then included my branch in `package.json`. I believe including `goodeggs/hubot-pager-me#fix-trigger-escalation-policy` will do the same.

Note that we had two separate problems: one was an incorrectly set `HUBOT_PAGERDUTY_FROM_EMAIL`, and the second was fixed by #130. They both resulted in the same error from PagerDuty (1008).
